include/rocksdb/table.h

```diff
@@ -100,6 +100,15 @@ struct BlockBasedTableOptions {
 
   IndexType index_type = kBinarySearch;
 
+  // The format type that will be used for the data block.
+  // Now two BlockFormatType supported
+  enum DataBlockIndexType : char {
+    kDataBlockBinarySearch = 0,  // traditional block type
+    kDataBlockHashIndex = 1,     // additional hash index appended to the end.
```

Comment:
Shall we rename this to kDataBlockHashSearch, to be similar to table IndexType ?

Reply:
How about rename `kDataBlockBinarySearch` to be `kDataBlockNoIndex`?
I struggled a little here to find a good name. As we are naming `DataBlockIndexType`, I tried to end with -Index.
BinarySearch has no index, so the index type may be called `kDataBlockNoIndex`. But maybe `kDataBlockBinarySearch` provides more information. What do you think?
Or, we can rename the enum type as `DataBlockSearchType`. But `IndexType` seems to be more informative.

Reply:
Oh I see the existing code use `kHashSearch` etc. in `IndexType`. So never mind, I will `s/kDataBlockHashIndex/kDataBlockHashIndex/`

Reply:
Yeah, thanks. I was just trying to see if we can name it to something similar to what users are already seeing as part of our public api. 
It should be ok to keep using hash index as the name for internal implementation.